File: Front/tpi-teg-grupo-11/src/app/core/models/class/turn.ts
import {Game} from './game';
import {TurnPhase} from '../../enums/turn-phase';
import {Player} from './player';
import {Movement} from './movement';

export interface Turn {
  turnId: number,
  game: Game,
  turnPhase: TurnPhase,
  player: Player,
  movements: Movement[]
}

File: Back/tpi-teg-grupo-11/src/main/java/ar/edu/utn/frc/tup/piii/models/Turn.java
package ar.edu.utn.frc.tup.piii.models;

import ar.edu.utn.frc.tup.piii.enums.TurnPhase;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;


@Data
@AllArgsConstructor
@NoArgsConstructor
public class Turn {
    private Long turnId;
    private UUID gameId;
    private TurnPhase turnPhase;
    private Player player;
    private List<Movement> movements;
    private Integer roundNumber ;
}
